src/main/kotlin/com/figure/gradle/semver/internal/util/MemoizedGit.kt

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2023 Figure Technologies and its affiliates.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE.md file in the root directory of this source tree.
+ */
+
+package com.figure.gradle.semver.internal.util
+
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import java.io.File
+
+object MemoizedGit {
+    private val memoizedGit = mutableMapOf<String, Git>()
```

Comment:
From an article:
>Memoization is a specific form of caching that is used in dynamic programming. The purpose of caching is to improve the performance of our programs and keep data accessible that can be used later. It basically stores the previously calculated result of the subproblem and uses the stored result for the same subproblem. This removes the extra effort to calculate again and again for the same problem. And we already know that if the same problem occurs again and again, then that problem is recursive in nature.

So in this case, I'm trying to avoid any slowdown there may be in creating the Git object and just fetching from what we're already aware of.